promql/engine.go

```diff
@@ -59,6 +59,9 @@ const (
 	maxInt64 = 9223372036854774784
 	// The smallest SampleValue that can be converted to an int64 without underflow.
 	minInt64 = -9223372036854775808
+
+	// Max initial size for the pooled points slices.
```

Comment:
I might expand this explanation a little - `getFPointSlice` and `getHPointSlice` are called with a guess as to the expected size, but often that guess is an over-estimate.